File: my_py_pkg/my_py_pkg/hardware_status_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus


class HardwareStatusPublisherNode(Node):
    def __init__(self):
        super().__init__("hardware_status_publisher")

        self.hw_status_pub_ = self.create_publisher(
            msg_type=HardwareStatus, topic="hardware_status", qos_profile=10
        )
        self.timer_ = self.create_timer(
            timer_period_sec=1, callback=self.publish_hw_status
        )

        self.get_logger().info("Hardware Status Publisher started.")

    def publish_hw_status(self):
        msg = HardwareStatus()
        msg.temperature = 43.7
        msg.are_motors_ready = True
        msg.debug_message = "Nothing special"
        self.hw_status_pub_.publish(msg)


def main(args=None):
    rclpy.init(args=args)

    node = HardwareStatusPublisherNode()
    rclpy.spin(node=node)

    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/hardware_status_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <my_robot_interfaces/msg/hardware_status.hpp>

using namespace std::chrono_literals;

class HardwareStatusPublisherNode : public rclcpp::Node
{
public:
  HardwareStatusPublisherNode()
      : Node("hardware_status_publisher")
  {
    pPublisher_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hardware_status", 10);
    pTimer_ = this->create_wall_timer(1s, std::bind(&HardwareStatusPublisherNode::publishHardwareStatus, this));

    RCLCPP_INFO(this->get_logger(), "Hardware Status Publisher started.");
  }

private:
  void publishHardwareStatus()
  {
    auto msg = my_robot_interfaces::msg::HardwareStatus();
    msg.temperature = 57.2;
    msg.are_motors_ready = false;
    msg.debug_message = "Motors are too hot!";
    pPublisher_->publish(msg);
  }

  rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr pPublisher_;
  rclcpp::TimerBase::SharedPtr pTimer_;
};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);

  auto pNode = std::make_shared<HardwareStatusPublisherNode>();
  rclcpp::spin(pNode);

  rclcpp::shutdown();
  return 0;
}
